### ISSUE TITLE
Nesting level too deep

### ISSUE THREAD

**Issue by JochenHWezel** [owncloud/core#25269]
The server log show following error:
Nesting level too deep - recursive dependency? at /var/www/owncloud/lib/private/files/view.php#729

I realized that currently, my windows owncloud client (most current version) can't sync some files to the server.

### Server configuration
**Operating system**: ubuntu 16.04

**Web server:** apache

**Database:** mysql

**PHP version:**

**ownCloud version:** OwnCloud v9.0.2

**Updated from an older ownCloud or fresh install:** Initially installed v9.0.x

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.2.2
  - comments: 0.2
  - dav: 0.1.6
  - documents: 0.12.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_antivirus: 0.8.0.1
  - files_external: 0.5.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - user_ldap: 0.8.0
Disabled:
  - encryption
  - external
  - user_external
```

**The content of config/config.php:**

```
{
    "system": {
        "updatechecker": false,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            ""***REMOVED SENSITIVE VALUE***",
            ""***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/var\/www\/owncloud_data",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "overwrite.cli.url": "https:\/\/"***REMOVED SENSITIVE VALUE***\/owncloud",
        "dbtype": "mysql",
        "version": "9.0.2.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_from_address": "no-reply",
        "mail_smtpmode": "smtp",
        "mail_domain": ""***REMOVED SENSITIVE VALUE***",
        "ldapIgnoreNamingRules": false,
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "smtp.1und1.de",
        "mail_smtpport": "25",
        "mail_smtpsecure": "tls",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 2
    }
}

```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** yes in meaning of https, no in meaning of file encryption

**Are you using an external user-backend, if yes which one:** no

#### ownCloud log (data/owncloud.log)
[owncloud.zip](https://github.com/owncloud/core/files/333478/owncloud.zip)

**Comment by RealRancor** [comment]
@JochenHWezel Do you have moved the files between two shares? If yes this is very likely a duplicate of https://github.com/owncloud/core/issues/24318 (found it via the search here at the repos) which gets fixed in the upcoming 9.0.3 released in 1 1/2 week.

**Comment by JochenHWezel** [comment]
maybe - i'll try/see in around 2 weeks ;-)

**Comment by JochenHWezel** [comment]
I moved the directory with the files again to another location, waited for resync, moved back again, resynced and voilà: the error disappeared.

**Comment by PVince81** [comment]
Yes, this is fixed in 9.0.3 RC1
